integration-tests/validation/src/testDebug/kotlin/com/android/designcompose/testapp/validation/AnimationMidpoints.kt

```diff
@@ -76,6 +79,41 @@ class AnimationMidpoints {
         recordAnimation("Variant")
     }
 
+    @Test
+    fun customVariantAnimation() {
+        // Because we're testing animation, we will manually advance the animation clock.
+        composeTestRule.mainClock.autoAdvance = false
+
+        val state = mutableStateOf(State.X)
+        val text = mutableStateOf("X")
+
+        composeTestRule.setContent {
+            CompositionLocalProvider(
+                LocalDesignDocSettings provides
+                    DesignDocSettings(
+                        useSquoosh = true,
+                        customVariantTransition = {
+                            SmartAnimateTransition(
+                                tween(
```

Comment:
For some reason the midpoint snapshot looks like it hasn't moved from the start of the animation. Does it show something more mid-pointy if you change the curve?

Reply:
Yea the cubic bezier curves start out slow and the midpoint test only progresses time by 0.1 before taking the midpoint screenshot. I'll change back to a spring.

Reply:
Sounds good - otherwise you might be able to use some bigger constants in the first two terms of the cubic bezier to make it do something right way.